Audit: Check all names for a given email address. Retain errors.

jzaefferer opened the pull request:
This replaces the hashed method with the hashedDetails method, which now
removes duplicate names.

Fixes #32
Closes #38
Ref #17

scottgonzalez commented on lib/signatures.js:
```diff
@@ -50,9 +50,16 @@ function getSignatures( key ) {
+		} )
+		.then( function( signatures ) {
+			return validate( signatures );
+		} );
```
This method no longer just returns errors as the name implies. This also has a lot of overlap with `getSignatures()`/`all()`; should probably consolidate the logic.

jzaefferer replied:
That's correct, I'm working on a replacement. Wanted to start by increasing test coverage on this method. I'm now rewriting the method return a different data structure. Once all uses are switched over, I'll remove the current `hashed` method.

I guess at this point the added tests are more interesting than the implementation changes...

scottgonzalez replied:
Ok, well the test seems fine.

jzaefferer commented:
Updated with a new `hashedDetails` method. From the last commit:

> This keeps using the email address at key, but then adds a list of names and a list of errors for each. Good entries have one name and no errors. Okay entries have mutliple names and no errors. Bad entries have one or more names and one or more errors.

Unless this has some terrible flaw, next I'm going to update other parts of the program to make use of this new method.

jzaefferer commented:
Updated once more, now its doing what its actually supposed to do, not just some intermediate step. Goes together with https://github.com/jquery/jquery-wp-content/pull/371

@scottgonzalez @arschmitz Could I have another review?

arschmitz commented:
LGTM

jzaefferer commented:
Updated to fix the case where a signature exists, but has errors. There was one PR that was indeed still invalid. I've also squashed everything into a single commit.

jzaefferer commented:
Merged and tagged as 0.5.0. If something goes wrong while I'm on vacation the next 7 days, @arschmitz has access for emergency ops.
